packages/web/components/Dashboard/XP.tsx

```diff
@@ -1,45 +1,250 @@
+        fillOpacity: 0.5,
+        strokeWidth: 0,
+      },
+    },
+  },
+};
+
+export const XP = (): React.ReactElement => {
+  const { user } = useUser();
+
+  const {
+    userTotalXP,
+    variationThisWeek,
+    variationLastWeek,
+    thisWeekXP,
+    lastWeekXP,
+    userWeeklyCred,
+  } = useUserXP(user?.ethereum_address || '');
+
+  return (
+    <>
+      <StatGroup mt={5} flex="0 0 50%">
+        <Stat mb={3}>
+          <StatLabel>This Week</StatLabel>
+          <StatNumber>{thisWeekXP}</StatNumber>
+          <StatHelpText>
+            <StatArrow type={variationThisWeek < 0 ? 'decrease' : 'increase'} />
+            {variationThisWeek}%
+          </StatHelpText>
+        </Stat>
+
+        <Stat mb={3} flex="0 0 50%">
+          <StatLabel>Last Week</StatLabel>
+          <StatNumber>{lastWeekXP}</StatNumber>
+          <StatHelpText>
+            <StatArrow type={variationLastWeek < 0 ? 'decrease' : 'increase'} />
+            {lastWeekXP}%
+          </StatHelpText>
+        </Stat>
+
+        <Stat alignSelf="flex-start" justifySelf="flex-end" flex="0 0 100%">
+          <StatLabel>All Time</StatLabel>
+          <StatNumber>{userTotalXP}</StatNumber>
+          {user?.player?.rank && (
+            <MetaTag
+              backgroundColor={user.player.rank.toLowerCase()}
+              size="md"
+              color="blackAlpha.600"
+            >
+              {user.player.rank}
+            </MetaTag>
+          )}
+        </Stat>
+        {/* Currently we are not building this stat  */}
+        {/* <Stat alignSelf="flex-start" justifySelf="flex-end" flex="0 0 50%">
+        <StatLabel>XP/SEED Ratio</StatLabel>
+        <StatNumber>3.4</StatNumber>
+        <StatHelpText>
+          <StatArrow type="decrease" />
+          0.3
+        </StatHelpText>
+      </Stat> */}
+      </StatGroup>
+      <Box
+        className="chartWrapper"
+        position="absolute"
+        width="100%"
+        height="100%"
+        bottom={0}
+        left={0}
+        zIndex={0}
+        sx={chartWrapperStyles}
+      >
+        {userWeeklyCred.length > 0 && <Chart data={userWeeklyCred} />}
+      </Box>
+    </>
+  );
+};
+
+type ChartType = {
+  children?: ReactNode;
+  data: Array<number>;
+};
+
+const ChartRange = ({ value = {} }): React.ReactElement => (
+  <Box>{`${value}d`}</Box>
 );
+const AnimatedChartRange = animated(ChartRange);
+
+export const Chart: FC<ChartType> = ({ data }) => {
+  const [scale, setScale] = useState<number>(30);
+  const props = useSpring({
+    config: {
+      mass: 10,
+      tension: 280,
+      friction: 100,
+    },
+    range: scale,
+  });
+
+  const switchScale = (range: number) => {
+    setScale(range);
+  };
+
+  function makePlots(weeks: Array<number>) {
+    const plots: Array<AreaSeriesPoint> = [];
+
+    weeks.slice(scale === 90 ? -12 : -4).map((d, i) => {
+      const week = {
+        x: i + 1,
+        y: d,
+        y0: 0,
+      };
+      return plots.push(week);
+    });
```

Comment:
```suggestion
    const plots = (
      weeks.slice(scale === 90 ? -12 : -4)
      .map((d, i) => ({
        x: i + 1,
        y: d,
        y0: 0,
      })
    );
```

Reply:
Also, since scale is the number of days, why not `slice(Math.floor(scale / 7))`?

Reply:
understood